020-Connect/010-Postgres/020-data-model.mdx

```diff
@@ -0,0 +1,40 @@
+---
+title: Data model
+description: Postgres types mapped to Xata types
+keywords: ['sql', 'mappings', 'type']
+slug: data-model
+published: true
+---
+
+Database tables are stored in PostgreSQL as tables. Xata data types are mapped to PostgreSQL types with additional constraints:
+
+| Xata data type | PostgreSQL data type | Notes                              |
+| -------------- | -------------------- | ---------------------------------- |
+| string         | text                 |                                    |
+| text           | text                 | With constraints on maximum length |
+| int            | bigint               |                                    |
+| float          | double precision     |                                    |
+| datetime       | timestamptz          |                                    |
+| bool           | boolean              |                                    |
+| email          | text                 | Validated by Xata                  |
+| multiple       | text[]               | Validated by Xata                  |
+| file           | jsonb                | Validated by Xata                  |
+| filearray      | jsonb                | Validated by Xata                  |
+| link           | foreign key          |                                    |
+
+In PostgreSQL tables are used to store and organize data. Linked columns serve as foreign keys, establishing connections to other tables. To access data from a linked column, use JOIN operations or apply WHERE conditions to locate the related records.
+
+For example, the following query shows how to retrieve email addresses and corresponding cities using a `LEFT JOIN` operation between the "Users" and "address" tables.:
+
+```sql
+SELECT "Users".email, address.city FROM "Users"
+LEFT JOIN address ON "Users".id=address.id;
+```
+
+For specific Xata data types like email, it is required you use the Xata API or SDK. This ensures that Xata can validate the inserted data's accuracy. We recommend refraining from directly writing to columns of this type and instead using the provided API for better results.
+
+## Limitations
+
+As you are able to create your own types or use types from extensions through the [wire protocol](/docs/postgres), there is a possibility that the Xata UI is not aware the type.
+For such cases, the UI will display the column values as string and they will be read-only.
+If you need to manipulate these columns, please do so through the wire protocol or reach out to our [support team](/docs/resources/support) to request adding the data type to Xata.
```

Comment:
```suggestion
If you need to manipulate these columns, please do so through the wire protocol or reach out to our [support team](/docs/resources/support) to request adding the data type to Xata.

Branch migrations are not supported yet. In databases that support direct Postgres access, you can create new branches but they cannot be merged together. This capability will become available later on.
```

@philkra adding a comment limitation for branch migrations that we'll want to remove later on when there is progress.